### DISCUSSION TITLE
Removal of plenary.nvim as a required dependency.

### DISCUSSION THREAD

**Discussion by 231tr0n** [olimorris/codecompanion.nvim#491] [Ideas]
Hi @olimorris! Thank you for creating this plugin. Is it in scope of codecompanion.nvim to remove plenary.nvim as a dependency and ship this plugin with only treesitter as the required dependency.

**Comment by olimorris** [comment]
Hey @231tr0n. Unfortunately not. I use Plenary for its Curl, Async and Path libraries. It should be a common plugin for nearly all Neovim users.

**Reply by 231tr0n** [reply]
@olimorris are you open to accepting PR to remove plenary.nvim as a dependency. If so, I can try hacking something out.

**Reply by olimorris** [reply]
It doesn't make sense to remove plenary and shift the maintenance burden into CodeCompanion, unfortunately.

As an aside, we've recently removed Tree-sitter as a dependency though (unless you need settings or the symbols slash command).

**Reply by 231tr0n** [reply]
Nvim-treesitter I use it to get parsers installed as it is the best way but plenary is the plugin I generally try to avoid since it is not as well maintained as other plugins I use.

**Reply by olimorris** [reply]
I agree it's not had the most activity recently. That being said, it's probably as feature complete as it needs to be
